int32 mode
bool enabled
int32 speed
int32 angle
